website/docs/r/cdn_endpoint_custom_domain.html.markdown

```diff
@@ -75,7 +75,31 @@ The following arguments are supported:
 
 * `host_name` - (Required) The host name of the custom domain. Changing this forces a new CDN Endpoint Custom Domain to be created.
 
-~> **Note:** At this time only HTTP Custom Domains are supported.
+* `cdn_managed_https` (Optional) - A `cdn_managed_https` block as defined below. 
+
+* `user_managed_https` (Optional) - A `user_managed_https` block as defined below.
+
+~> **NOTE** Only one of `cdn_managed_https` and `user_managed_https` can be specified.
+
+!> **Warning** It is allowed to update the HTTPS settings on the CDN Endpoint Custom Domain only by toggling it. In-place update on HTTPS settings is not allowed. This is because setting different HTTPS settings will need a disable-then-enable process. When HTTPS settings got disabled, the service will take ~8 hours to clean up your previous enablement request for the same custom domain and there is no way to get notification when that clean up has done.
```

Comment:
Although the detailed explanation here is useful, I think that this level of detail belongs as a comment in the code (perhaps near the switch block in update) and would shorten this to:
```suggestion
!> **Warning** Due to the way the Azure API handles in-place updates on HTTPS settings for the Custom Domain of a CDN Endpoint, it is only possible to toggle these settings between disabled and enabled individually.
```